Source/DataSources/EntityCluster.js

```diff
@@ -0,0 +1,775 @@
+        hasLabel = hasLabel && (!defined(labelGraphics._font) || Property.isConstant(labelGraphics._font));
+        hasLabel = hasLabel && (!defined(labelGraphics._style) || Property.isConstant(labelGraphics._style));
+        hasLabel = hasLabel && (!defined(labelGraphics._fillColor) || Property.isConstant(labelGraphics._fillColor));
+        hasLabel = hasLabel && (!defined(labelGraphics._outlineColor) || Property.isConstant(labelGraphics._outlineColor));
+        hasLabel = hasLabel && (!defined(labelGraphics._outlineWidth) || Property.isConstant(labelGraphics._outlineWidth));
+
+        if (hasLabel) {
+            var label = entityCluster._clusterLabelCollection.add();
+
+            label.show = true;
+            label.position = position;
+            label.text = Property.getValueOrUndefined(labelGraphics._text, undefined);
+            label.font = Property.getValueOrDefault(labelGraphics._font, undefined, defaultFont);
+            label.style = Property.getValueOrDefault(labelGraphics._style, undefined, defaultStyle);
+            label.fillColor = Property.getValueOrDefault(labelGraphics._fillColor, undefined, defaultFillColor);
+            label.outlineColor = Property.getValueOrDefault(labelGraphics._outlineColor, undefined, defaultOutlineColor);
+            label.outlineWidth = Property.getValueOrDefault(labelGraphics._outlineWidth, undefined, defaultOutlineWidth);
+
+            label.id = ids;
+        }
+
+        var billboardGraphics = entity._billboard;
+        if (defined(billboardGraphics) && defined(billboardGraphics._image) && Property.isConstant(billboardGraphics._image)) {
+            var billboard = entityCluster._clusterBillboardCollection.add();
+
+            billboard.show = true;
+            billboard.position = position;
+            billboard.image = billboardGraphics._image.getValue();
+
+            billboard.id = ids;
+        }
+    }
+
+    function getScreenSpacePositions(collection, points, scene, occluder) {
+        if (!defined(collection)) {
+            return;
+        }
+
+        var length = collection.length;
+        for (var i = 0; i < length; ++i) {
+            var item = collection.get(i);
+            item._clusterRender = false;
+
+            if (!item.show || !occluder.isPointVisible(item.position)) {
+                continue;
+            }
+
+            if (defined(item._labelCollection) && (defined(item.id._billboard) || defined(item.id._point))) {
+                continue;
+            }
+
+            var coord = item.computeScreenSpacePosition(scene);
+            if (!defined(coord)) {
+                continue;
+            }
+
+            points.push({
+                index : i,
+                collection : collection,
+                clustered : false,
+                coord : coord
+            });
+        }
+    }
+
+    function createDeclutterCallback(entityCluster) {
```

Comment:
Is it possible to move this clustering algorithm into `Core` as a `@private` class/function?